src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -49,6 +52,75 @@ public static Encoding OutputEncoding
             get { return GetConsoleEncoding(); }
         }
 
+        private static readonly object InternalSyncObject = new object();
```

Comment:
We will be synchronizing against things not really required. So, I would prefer keeping a separate one if you don't feel strongly about it.